docs/Segments.md

```diff
@@ -118,6 +118,96 @@ This is platform specific; parses the data and interprets as a header for e.g. N
   start: 0xABC
 ```
 
+## `data`
+
+**Description:**
+
+Data located in the ROM.
+
+**Example:**
+
+```yaml
+# as list
+- [0xABC, data, filepath]
+
+# as dictionary
+- name: filepath
+  type: data
+  start: 0xABC
+```
+
+This will created `filepath.asm` in your `asm` folder.
+
+## `.data`
+
+**Description:**
+
+Data located in the ROM, linked from a C file.
+
+Once you have figured out the types of symbols in the data section and you are confident about its file split, you will want probably to migrate symbols from assembly to C. To do this, it's possible to prefix the `data` with a `.`, like `.data`. This instructs the linker to, in the build stage, link to the symbols in the C file specified at `filepath`. You'll have to define the symbols in this C file to prevent undefined reference errors.
+
+`splat` will not generate `.data.s` files for these sections, as the symbols should be declared in the specified C file.
+
+
+**Example:**
+
+```yaml
+# as list
+- [0xABC, .data, filepath]
+
+# as dictionary
+- name: filepath
+  type: .data
+  start: 0xABC
+```
+
+This will created `filepath.c` in your `src` folder where you can define the symbols which were originally defined in the `data` segment.
+
+## `rodata`
+
+**Description:**
+
+Read-only data located in the ROM.
+
+**Example:**
+
+```yaml
+# as list
+- [0xABC, rodata, filepath]
+
+# as dictionary
+- name: filepath
+  type: rodata
+  start: 0xABC
+```
+
+This will created `filepath.s` in your `asm` folder.
+
+## `.rodata`
+
+**Description:**
+
+Read-only data located in the ROM, linked to a C file.
+
+If you want to migrate symbols from assembly to C, please prefix the `rodata` with a `.`, like `.rodata`. 
+
+`splat` will not generate `.rodata.s` files for these sections, as the symbols should be declared in the specified C file.
+
+**Example:**
+
+```yaml
+# as list
+- [0xABC, .rodata, filepath]
+
+# as dictionary
+- name: filepath
+  type: .rodata
+  start: 0xABC
+```
+
+This will created `filepath.c` in your `src` folder where you can define the symbols which were originally defined in the `rodata` segment.
```

Comment:
```suggestion
This will create `filepath.c` in your `src` folder where you can define the symbols which were originally defined in the `rodata` segment.
```